### PR TITLE
do not lie about the strings encoding

### PR DESCRIPTION
**Pull request by james-lawrence**

### CODE REVIEW

File: lib/mail/version_specific/ruby_1_9.rb [hunk 1]
```diff
@@ -89,7 +89,7 @@ def Ruby19.b_value_decode(str)
       decoded.valid_encoding? ? decoded : decoded.encode(Encoding::UTF_16LE, :invalid => :replace, :replace => "").encode(Encoding::UTF_8)
     rescue Encoding::UndefinedConversionError, ArgumentError, Encoding::ConverterNotFoundError
       warn "Encoding conversion failed #{$!}"
-      str.dup.force_encoding(Encoding::UTF_8)
+      str.dup
```
Comment (mwooddatto) [inline lib/mail/version_specific/ruby_1_9.rb:91]: So without the encoding forced, this will allow exceptions to be thrown elsewhere as needed, right?

**Comment by rfleisher** [comment]
You made this change in presumably the file that handles ruby version 1.9. Does the Mail gem use the 1.9 ruby version file for later versions? I see they have some compatible version for 2.1.2.

**Comment by james-lawrence** [comment]
not in the 2.6.3 branch they dont (which is what we are using)

File: spec/mail/encodings_spec.rb [hunk 2]
```diff
@@ -133,7 +133,7 @@
     it "should decode a long encoded string" do
       string = '=?UTF-8?B?VGhpcyBpcyDjgYIgcmVhbGx5IGxvbmcgc3RyaW5nIFRoaXMgaXMg44GCIHJl?= =?UTF-8?B?YWxseSBsb25nIHN0cmluZyBUaGlzIGlzIOOBgiByZWFsbHkgbG9uZyBzdHJp?= =?UTF-8?B?bmcgVGhpcyBpcyDjgYIgcmVhbGx5IGxvbmcgc3RyaW5nIFRoaXMgaXMg44GC?= =?UTF-8?B?IHJlYWxseSBsb25nIHN0cmluZw==?='
       result = "This is あ really long string This is あ really long string This is あ really long string This is あ really long string This is あ really long string"
-      result.force_encoding('UTF-8') if RUBY_VERSION >= '1.9'
+      result.encode('UTF-8') if RUBY_VERSION >= '1.9'
```
Comment (akcrono) [inline spec/mail/encodings_spec.rb:136]: Don't you need encode! to change the original string?

```
[4] pry(main)> temp = 'how are you?'
=> "how are you?"
[5] pry(main)> temp.encode 'ascii-8bit'
=> "how are you?"
[6] pry(main)> temp.encoding
=> #<Encoding:UTF-8>
[7] pry(main)> temp.encode! 'ascii-8bit'
=> "how are you?"
[8] pry(main)> temp.encoding
=> #<Encoding:ASCII-8BIT>
[9] pry(main)>
```

**Pull request closed without merging**